### DISCUSSION TITLE
exclude specific dependency

### DISCUSSION THREAD

**Discussion by electriquo** [renovatebot/renovate#23927] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

exclude specific dependency from being updated

### What you tried so far.

looked at the logs in the "packageFiles with updates" section to identify the properties of the dependency. i noted the package manager (pre-commit), datasource (github-tags), dependency name (depName, foo/baz) and the package name (packageName, foo/baz). then tried to add each of the following packageRules in renovate configuration

```json
  "packageRules": [
    {
      "matchPackageNames": ["foo/baz"],
      "matchManagers": ["pre-commit"],
      "matchDatasources": ["github-tags"],
      "enabled": false,
    },
  ]
```
also tried

```json
  "packageRules": [
    {
      "excludeDepNames": ["foo/baz"],
      "matchManagers": ["pre-commit"],
      "matchDatasources": ["github-tags"],
    },
  ]
```
also

```json
  "packageRules": [
    {
      "excludePackageNames": ["foo/baz"],
      "matchManagers": ["pre-commit"],
      "matchDatasources": ["github-tags"],
    },
  ]
```
but non of them work.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
Docker is not a valid package manager name and it looks like you should use matchPackageNames not patterns

**Reply by electriquo** [reply]
my bad, the match manager is pre-commit. i'll update accordingly.
although pre-commit is not officially supported. it works fine excluding specific dependencies that does not conform to semver schema. thus, would like to exclude them

**Comment by rarkins** [comment, marked as answer]
To exclude a specific dependency from being updated, do either:
- Add it to `ignoreDeps`
- Add a packageRule with `matchPackageNames` plus `enable=false`

**Reply by jblsk** [reply]
I am trying to use `ignoreDeps` in my packageRule, but that's not working for me. Could you please suggest an update to my config below?
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "commitMessagePrefix": "JIRA-KEY",
  "enabledManagers": ["kubernetes"],
  "fetchChangeLogs": "off",
  "kubernetes": {
    "fileMatch": ["^bootstrap/argo-cd/base/wave1/install\\.yaml$"]
  },
  "packageRules": [
    {
      "ignoreDeps": ["ghcr.io/dexidp/dex", "haproxy", "redis"],
      "matchDatasources": ["docker", "kubernetes"],
      "matchDepNames": ["quay.io/argoproj/argocd"],
    },
  ],
  "branchConcurrentLimit": 5,
  "prConcurrentLimit": 5,
  "reviewers": ["team:az-redacted"]
}
```

**Reply by RahulGautamSingh** [reply]
`ignoreDeps` should not be inside `packageRules`, it is a top-level config.

**Reply by jblsk** [reply]
Indeed, it passed the config validation, thank you. Now it finally works as expected.
